packages/caliper-fabric/lib/FabricConnectorFactory.js

```diff
@@ -26,46 +26,73 @@ const NEW_V1_WALLET_FACADE_FACTORY = './connector-versions/v1/WalletFacadeFactor
-    } else {
+        if (sdk) {
+            throw new Error('Another binding packages is already installed, pls unbind first before binding to a different fabric version');
+        }
+        packageVersion = semver.coerce(require('fabric-network/package').version);
+        sdk = 'fabric-network';
+    }
+
+
+    if (!sdk) {
         throw new Error('Unable to detect required Fabric binding packages');
     }
+
+    return {sdk, packageVersion};
 };
 
-const _loadAppropriateConnectorClass = (installedNodeSDKVersion) => {
+const _loadAppropriateConnectorClass = (installedNodeSdk, version) => {
     let connectorPath;
     let walletFacadeFactoryPath;
 
-    if (semver.satisfies(installedNodeSDKVersion, '=1.x')) {
-        const useGateway = ConfigUtil.get(ConfigUtil.keys.Fabric.Gateway.Enabled, false);
-        Logger.info(`Initializing ${useGateway ? 'gateway' : 'standard' } connector compatible with installed SDK: ${installedNodeSDKVersion}`);
+    if (installedNodeSdk ===  'fabric-network') {
+        if (semver.satisfies(version, '=1.x')) {
+            const useGateway = ConfigUtil.get(ConfigUtil.keys.Fabric.Gateway.Enabled, false);
+            Logger.info(`Initializing ${useGateway ? 'gateway' : 'standard' } connector compatible with installed fabric-network SDK: ${version}`);
 
-        if (!useGateway) {
-            connectorPath = NEW_V1_NODE_CONNECTOR;
-            walletFacadeFactoryPath = NEW_V1_WALLET_FACADE_FACTORY;
-        } else {
-            // gateway with default event handlers appears in SDK > 1.4.2
-            if (semver.satisfies(installedNodeSDKVersion, '>=1.4.2')) {
-                connectorPath = NEW_V1_GATEWAY_CONNECTOR;
+            if (!useGateway) {
+                connectorPath = NEW_V1_NODE_CONNECTOR;
                 walletFacadeFactoryPath = NEW_V1_WALLET_FACADE_FACTORY;
             } else {
-                throw new Error('Caliper currently only supports Fabric gateway based operation using Fabric-SDK 1.4.2 and higher. Please retry with a different SDK binding');
+                // gateway with default event handlers appears in SDK > 1.4.2
+                if (semver.satisfies(version, '>=1.4.2')) {
+                    connectorPath = NEW_V1_GATEWAY_CONNECTOR;
+                    walletFacadeFactoryPath = NEW_V1_WALLET_FACADE_FACTORY;
+                } else {
+                    throw new Error('Caliper currently only supports Fabric gateway based operation using Fabric-SDK 1.4.2 and higher. Please retry with a different SDK binding');
+                }
             }
+        } else if (semver.satisfies(version, '=2.x')) {
+            Logger.info(`Initializing gateway connector compatible with installed SDK: ${version}`);
+            connectorPath = NEW_V2_GATEWAY_CONNECTOR;
+            walletFacadeFactoryPath = NEW_V2_WALLET_FACADE_FACTORY;
+        } else {
+            throw new Error(`Installed SDK version ${version} did not match any compatible Fabric connectors`);
+        }
```

Comment:
We should just do a return here and then we can remove the following `else if` line, just a comment to say it must be a peer-gateway sdk